src/NUnitConsole/nunit3-console/ConsoleRunner.cs

```diff
@@ -169,7 +169,17 @@ private int RunTests(TestPackage package, TestFilter filter)
             foreach (var spec in _options.ResultOutputSpecifications)
             {
                 var outputPath = Path.Combine(_workDirectory, spec.OutputPath);
-                GetResultWriter(spec).CheckWritability(outputPath);
+                try
+                {
+                    GetResultWriter(spec).CheckWritability(outputPath);
+                }
+                catch (SystemException ex)
+                {
+                    throw new NUnitEngineException(
```

Comment:
@CharliePoole IMO throwing NUnitEngine or any Exception here without suppressing the stacktrace in Program.cs is not improving the readability of the error message because the resultant exception looks like this, And once you read the error message in the first line the stacktrace is basically redundant.

```
C:\Users\vignesh.rajasekaran3\gitspace\nunit-console\bin\Debug [master]
λ .\nunit3-console.exe .\mock-assembly.dll --work=c:\windows\
NUnit Console Runner 3.5.0 (Debug)
Copyright (C) 2016 Charlie Poole

Runtime Environment
   OS Version: Microsoft Windows NT 10.0.14393.0
  CLR Version: 4.0.30319.42000

Test Files
    .\mock-assembly.dll

NUnit.Engine.NUnitEngineException: The path specified in --result TestResult.xml could not be written to ---> System.UnauthorizedAccessException: Access to the path 'c:\windows\TestResult.xml' is denied.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.StreamWriter.CreateFile(String path, Boolean append, Boolean checkHost)
   at System.IO.StreamWriter..ctor(String path, Boolean append, Encoding encoding, Int32 bufferSize, Boolean checkHost)
   at System.IO.StreamWriter..ctor(String path, Boolean append, Encoding encoding)
   at NUnit.Engine.Services.NUnit3XmlResultWriter.WriteResultFile(XmlNode resultNode, String outputPath) in c:\Users\vignesh.rajasekaran3\gitspace\nunit-console\src\NUnitEngine\nunit.engine\Services\ResultWriters\NUnit3XmlResultWriter.cs:line 53
   at NUnit.Engine.Services.NUnit3XmlResultWriter.CheckWritability(String outputPath) in c:\Users\vignesh.rajasekaran3\gitspace\nunit-console\src\NUnitEngine\nunit.engine\Services\ResultWriters\NUnit3XmlResultWriter.cs:line 48
   at NUnit.ConsoleRunner.ConsoleRunner.RunTests(TestPackage package, TestFilter filter) in c:\Users\vignesh.rajasekaran3\gitspace\nunit-console\src\NUnitConsole\nunit3-console\ConsoleRunner.cs:line 174
   --- End of inner exception stack trace ---
   at NUnit.ConsoleRunner.ConsoleRunner.RunTests(TestPackage package, TestFilter filter) in c:\Users\vignesh.rajasekaran3\gitspace\nunit-console\src\NUnitConsole\nunit3-console\ConsoleRunner.cs:line 178
   at NUnit.ConsoleRunner.ConsoleRunner.Execute() in c:\Users\vignesh.rajasekaran3\gitspace\nunit-console\src\NUnitConsole\nunit3-console\ConsoleRunner.cs:line 129
   at NUnit.ConsoleRunner.Program.Main(String[] args) in c:\Users\vignesh.rajasekaran3\gitspace\nunit-console\src\NUnitConsole\nunit3-console\Program.cs:line 129
```

Whilst I think what could make it easier to identify this issue is an output something like this. I'm thinking a little bigger here in terms of throwing the exception only when trying to write the results into the file. Basically separating of concerns of running the tests and saving the results. Not sure if this is aligning with the design but just my 2 cents 😄 

```
C:\Users\vignesh.rajasekaran3\gitspace\nunit-console\bin\Debug [master]
λ .\nunit3-console.exe .\mock-assembly.dll --work=c:\windows\
NUnit Console Runner 3.5.0 (Debug)
Copyright (C) 2016 Charlie Poole

Runtime Environment
   OS Version: Microsoft Windows NT 10.0.14393.0
  CLR Version: 4.0.30319.42000

Test Files
    .\mock-assembly.dll

<results of tests actually run>

Run Settings
    DisposeRunners: True
    WorkDirectory: c:\windows\
    ImageRuntimeVersion: 2.0.50727
    ImageRequiresX86: False
    ImageRequiresDefaultAppDomainAssemblyResolver: False
    NumberOfTestWorkers: 8

Test Run Summary
  Overall result: Failed
  Test Count: 28, Passed: 15, Failed: 5, Inconclusive: 1, Skipped: 7
    Failed Tests - Failures: 1, Errors: 1, Invalid: 3
    Skipped Tests - Ignored: 4, Explicit: 3, Other: 0
  Start time: 2016-10-27 04:16:03Z
    End time: 2016-10-27 04:16:03Z
    Duration: 0.504 seconds

Unable to save results to c:\windows\TestResults.xml due to lack of permission.
```